index.d.ts

```diff
@@ -302,7 +302,7 @@ declare module '@primer/components' {
 
   export interface SelectMenuListProps extends CommonProps, Omit<React.HTMLAttributes<HTMLDivElement>, 'color'> {}
 
-  export interface SelectMenuItemProps extends Omit<CommonProps, 'as'>,
+  export interface SelectMenuItemProps extends CommonProps,
     Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'color'> {
```

Comment:
I wonder if this needs to take into account that this could be an `<HTMLButtonElement>` too? The `as` prop is a little weird with TS to begin with so it's probably ok to leave how it is - if we hit issues we can dig deeper - I don't really know what the correct answer would be off the top of my head.

Reply:
Oof yeah good question 🤔

Reply:
@smockle this is the definition we're trying to figure out!

Reply:
Opened https://github.com/primer/components/pull/794 to explore this. Details in that PR description. I’d like to test the new types before we merge—I’ll follow-up here by EOD today.

Reply:
The [`does not allow the "as" prop on SelectMenu.Item` test (`SelectMenu.js#L77-L85`)](https://github.com/primer/components/blob/17108d4edeed0b0c45165b97f9cc5fe08885cd3d/src/__tests__/SelectMenu.js#L77-L85) is failing. The test ought to be updated in light of this PR, right? IIUC, we now _want_ to allow `"as"` on `SelectMenu.Item`.

Reply:
Yep, that's right!